fenix/app/src/main/java/org/mozilla/fenix/shopping/store/ReviewQualityCheckAction.kt

```diff
@@ -105,15 +110,55 @@ sealed interface ReviewQualityCheckAction : Action {
     /**
      * Triggered when the user clicks on learn more link on the opt in card.
      */
-    object OpenOnboardingLearnMoreLink : NavigationMiddlewareAction
+    object OpenOnboardingLearnMoreLink : NavigationMiddlewareAction, TelemetryAction
 
     /**
      * Triggered when the user clicks on terms and conditions link on the opt in card.
      */
-    object OpenOnboardingTermsLink : NavigationMiddlewareAction
+    object OpenOnboardingTermsLink : NavigationMiddlewareAction, TelemetryAction
 
     /**
      * Triggered when the user clicks on privacy policy link on the opt in card.
      */
-    object OpenOnboardingPrivacyPolicyLink : NavigationMiddlewareAction
+    object OpenOnboardingPrivacyPolicyLink : NavigationMiddlewareAction, TelemetryAction
+
+    /**
+     * Triggered when the bottom sheet is closed.
+     */
+    object BottomSheetClosed : TelemetryAction
+
+    /**
+     * Triggered when the bottom sheet is opened.
+     */
+    object BottomSheetDisplayed : TelemetryAction
+
+    /**
+     * Triggered when the user clicks on the "Not now" button from the contextual onboarding card.
+     */
+    object NotNowClicked : TelemetryAction
+
+    /**
+     * Triggered when the user expands the recent reviews card.
+     */
+    object ShowMoreRecentReviewsClicked : TelemetryAction
+
+    /**
+     * Triggered when the user expands the settings card.
+     */
+    object ExpandSettingsClicked : TelemetryAction
+
+    /**
+     * Triggered when the No analysis card is displayed to the user.
+     */
+    object NoAnalysisDisplayed : TelemetryAction
+
+    /**
+     * Triggered when the user clicks on the analyze button
+     */
+    object AnalyzeClicked : TelemetryAction
```

Comment:
So seems like we want to be specific about analysis and reanalysis. So let's call it `AnalyzeProduct` similar to `ReanalyzeProduct`. But we should also update `NoAnalysis` composable to call this action instead via passed function and also make it a `NetworkAction` and `UpdateAction` similar to `ReanalyzeProduct` so the same side effects in network middleware and reducer also occur on this action.